src/batch_triangular/Xblas_core.ch

```diff
@@ -0,0 +1,67 @@
+/**
+  - -* (C) Copyright 2013 King Abdullah University of Science and Technology
+  Authors:
+  Ali Charara (ali.charara@kaust.edu.sa)
+  David Keyes (david.keyes@kaust.edu.sa)
+  Hatem Ltaief (hatem.ltaief@kaust.edu.sa)
+
+  Redistribution  and  use  in  source and binary forms, with or without
+  modification,  are  permitted  provided  that the following conditions
+  are met:
+
+  * Redistributions  of  source  code  must  retain  the above copyright
+  * notice,  this  list  of  conditions  and  the  following  disclaimer.
+  * Redistributions  in  binary  form must reproduce the above copyright
+  * notice,  this list of conditions and the following disclaimer in the
+  * documentation  and/or other materials provided with the distribution.
+  * Neither  the  name of the King Abdullah University of Science and
+  * Technology nor the names of its contributors may be used to endorse
+  * or promote products derived from this software without specific prior
+  * written permission.
+  *
+  THIS  SOFTWARE  IS  PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  ``AS IS''  AND  ANY  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED  TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A  PARTICULAR  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDERS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL,  EXEMPLARY,  OR  CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT NOT
+  LIMITED  TO,  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA,  OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY  OF  LIABILITY,  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF  THIS  SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**/
+#ifndef __XBLAS_CORE__
+#define __XBLAS_CORE__
+
+
+#include "kblas_struct.h"
+#include "kblas_prec_def.h"
+
+//==============================================================================================
+void Xgemm_batch_strided_wsquery(int batchCount, kblasWorkspace_t ws);
+
+int Xgemm_batch_strided(kblasHandle_t handle,
```

Comment:
compliant with state of the art API?

Reply:
this is an internal wrapper function (among many others to come) not to be exposed as kblas API. X denotes a templated / overloaded function.